### PR TITLE
removed progressMade() from the callback.  Progress events only occur…

### PR DESCRIPTION
**Pull request by john-hill**
… from the heartbeat

### CODE REVIEW

File: src/main/java/org/sagebionetworks/common/util/progress/ProgressListener.java [hunk 1]
```diff
@@ -2,15 +2,19 @@
...
-public interface ProgressListener<T> {
+public interface ProgressListener {
 
 	/**
 	 * Called when progress is made.
 	 * 
-	 * @param t
+	 * Note: Progress events will be fired on a separate thread from the thread
+	 * the worker is running on. This allows progress events to be generated
+	 * even when the worker thread is block. Therefore, thread safety must be
```
Comment (john-hill) [inline src/main/java/org/sagebionetworks/common/util/progress/ProgressListener.java]: blocked

File: src/main/java/org/sagebionetworks/common/util/progress/ProgressListener.java [hunk 2]
```diff
@@ -2,15 +2,19 @@
...
+	 * exercised for all objects access by both the progress listener and
+	 * worker. Also, progress listeners cannot participate in database
+	 * transactions started by the worker, and vis versa.
```
Comment (john-hill) [inline src/main/java/org/sagebionetworks/common/util/progress/ProgressListener.java]: vice versa

File: src/test/java/org/sagebionetworks/common/util/progress/ThrottlingProgressCallbackTest.java [hunk 3]
```diff
@@ -13,7 +13,7 @@
 public class ThrottlingProgressCallbackTest {
```
Comment (john-hill) [inline src/test/java/org/sagebionetworks/common/util/progress/ThrottlingProgressCallbackTest.java]: Need to remove the Throttle now that the heartbeat is the only call of progress made.

File: src/main/java/org/sagebionetworks/common/util/progress/AutoProgressingCallable.java [hunk 4]
```diff
@@ -78,9 +80,17 @@ public R call() throws Exception {
...
-				callback.progressMade(null);
+				callback.fireProgressMade();
 			}
 		}
 	}
 
+	@Override
+	public R call(ProgressCallback callback) throws Exception {
+		if(!(callback instanceof SynchronizedProgressCallback)){
+			throw new IllegalArgumentException("ProgressCallback must extend AbstractProgressCallback");
```
Comment (xschildw) [inline src/main/java/org/sagebionetworks/common/util/progress/AutoProgressingCallable.java:91]: SynchronizedCallback

**Pull request merged**